holy-grail-backend/backend/app/api/endpoints/library.py

```diff
@@ -15,25 +15,39 @@
 notes_router = APIRouter()
 
 
-@router.post("/", response_model=NoteSchema)
+@router.post("/", response_model=list[NoteSchema])
```

Comment:
Also, don't put the business and service logic in the endpoint layer, make a custom `classmethod` in your `Library` model called `create_multiple_notes` or something then do this logic there. 

The code here in the endpoint should just call the method and pass the `request` and `session` parameters in and that's it.

Also you need to handle for `document_name` that already exists, if one of the docs has name that already exists, make sure all the files passes the condition of upload before doing a batch upload.